src/app/modules/recipe-form/components/products-chips/products-chips.component.ts

```diff
@@ -0,0 +1,143 @@
+import {COMMA, ENTER} from '@angular/cdk/keycodes';
+import {Component, ElementRef, EventEmitter, OnInit, Output, ViewChild} from '@angular/core';
+import {FormControl} from '@angular/forms';
+import {MatAutocompleteSelectedEvent} from '@angular/material/autocomplete';
+import {MatChipInputEvent} from '@angular/material/chips';
+import {ProductService} from "../../../products/services/product.service";
+import {Product} from "../../../../models/recipe-form.model";
+import { units } from '../../recipe-form.constants';
+
+@Component({
+  selector: 'product-chips',
+  templateUrl: './products-chips.component.html',
+  styleUrls: ['./products-chips.component.scss']
+})
+export class ProductsChipsComponent implements OnInit {
+
+  separatorKeysCodes: number[] = [ENTER, COMMA];
+  productCtrl = new FormControl();
+  productCountCtrl = new FormControl();
+  allProducts: Product[] = [];
+  products: Product[] = [];
+  filteredProducts: Product[] = [];
+  amounts: string[] = [];
+  units = units;
+  @Output() productsSubmit = new EventEmitter<Product[]>();
+  @ViewChild('productInput') productInput: ElementRef<HTMLInputElement> | undefined;
+
+  constructor(private productService: ProductService) {
+  }
+
+  updateLists(): void {
+    this.products = this.allProducts.filter(product => {
+      return product.amount != 0;
+    });
+    this.filteredProducts = this.allProducts.filter(product => product.amount === 0);
+    this.productsSubmit.emit(this.products);
+  }
+
+  ngOnInit(): void {
+    this.productService.getAll().subscribe(({data: products}) => {
+      this.allProducts = products.map(product => Object.assign({}, {name: product.name, id: product.id, amount: 0, unit:'Jednostka'}))
+    })
+    this.productCountCtrl.setValue(1);
+  }
+
+  add(event: MatChipInputEvent): void {
+    this.productCountCtrl.setValue(1);
+    const value = (event.value || '').trim();
+
+    if (value != '') {
+      if (this.filteredProducts.length === 1) {
+        this.filteredProducts[0].amount = 1;
+        event.chipInput!.clear();
+        this.updateLists();
+
+      }
+      else {
+        const productCheck = this.allProducts.find(product => product.name === value);
+        if (productCheck && productCheck.amount === 0) {
+          productCheck.amount = 1;
+          event.chipInput!.clear();
+          this.updateLists();
+        }
+      }
+    }
+  }
```

Comment:
try to avoid nesting ifs
```suggestion
  add(event: MatChipInputEvent): void {
    this.productCountCtrl.setValue(1);
    const value = (event.value || '').trim();

    if (value === '') {
      return;
    }
    if (this.filteredProducts.length === 1) {
      this.filteredProducts[0].amount = 1;
      event.chipInput!.clear();
      this.updateLists();
      return;
    }
    const productCheck = this.allProducts.find(product => product.name === value);
    if (productCheck && productCheck.amount === 0) {
      productCheck.amount = 1;
      event.chipInput!.clear();
      this.updateLists();
    }
  }
```